### PR TITLE
Add testnet option - Closes #148

### LINKED ISSUE #148: Add option for testnet true/false
For `get`, `list` and others going forward.

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #148 
~~IMPORTANT: functionality is not as expected because `setTestnet` has odd behaviour in lisk-js 0.5.0 and below. lisk-js has compatible behaviour in 1.0.0.~~
~~Merge https://github.com/LiskHQ/lisky/pull/366 first~~

### CODE REVIEW

**Comment by Tosch110** [comment]
`functionality is not as expected because` - maybe then we should wait before we implement it? Before people accidentally post things on the wrong network.

**Comment by willclarktech** [comment]
This PR is ready for review buy `suspended` until the `0.4.0` branch rebase onto `0.3.0` is finalised.

File: src/utils/query.js [hunk 1]
```diff
@@ -19,27 +19,43 @@ class Query {
...
 		};
 	}
 
-	getBlock(input) {
-		return this.client.sendRequest('blocks/get', { id: input });
+	sendRequest(endpoint, parameters, { testnet } = {}) {
+		const originalTestnetSetting = this.client.testnet;
+		const overrideTestnetSetting =
+			typeof testnet === 'boolean' && this.client.testnet !== testnet;
+		const reset = result => {
```
Comment (Tosch110) [inline src/utils/query.js]: The testnet settings (original setting and reset) seem to be good candidates for extra functions in the class.

Comment (willclarktech) [reply src/utils/query.js]: Updated.

File: test/specs/commands/list.js [hunk 2]
```diff
@@ -25,191 +25,193 @@ describe('list command', () => {
...
-					);
-				});
-				Given('a type "delegate"', given.anUnknownType, () => {
-					Given('inputs "lightcurve" and "tosch"', given.inputs, () => {
-						When(
-							'the action is called with the type and the inputs',
-							when.theActionIsCalledWithTheTypeAndTheInputs,
+						});
+					});
+					Given('a type "account"', given.anUnknownType, () => {
```
Comment (Tosch110) [inline test/specs/commands/list.js]: description does not match function name.

File: test/specs/commands/list.js [hunk 3]
```diff
@@ -25,191 +25,193 @@ describe('list command', () => {
...
 							},
 						);
 					});
-				});
-				Given('a type "delegates"', given.anUnknownType, () => {
-					Given('inputs "lightcurve" and "tosch"', given.inputs, () => {
-						When(
-							'the action is called with the type and the inputs',
-							when.theActionIsCalledWithTheTypeAndTheInputs,
+					Given('a type "accounts"', given.anUnknownType, () => {
```
Comment (Tosch110) [inline test/specs/commands/list.js]: as above. Some more of these in the tests

**Pull request merged**